File: client/src/app/modules/quotation/services/record.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { RecordType } from '../../quotation/model/RecordType';

@Injectable({
  providedIn: 'root'
})
export class RecordTypeService extends AppRestService<RecordType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getRecordTypes() {
    return this.getListCached(new HttpParams(), "record-types");
  }

  addOrUpdateRecordType(recordType: RecordType) {
    this.clearListCache(new HttpParams(), "record-types");
    return this.addOrUpdate(new HttpParams(), "record-type", recordType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/RecordTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.RecordType;

public interface RecordTypeService {
    public List<RecordType> getRecordTypes();

    public RecordType getRecordType(Integer id);

    public RecordType addOrUpdateRecordType(RecordType recordType);
}
